component/org.wso2.carbon.identity.casque.authenticator/src/main/java/org/wso2/carbon/identity/casque/authenticator/authenticator/CasqueAuthenticator.java

```diff
@@ -0,0 +1,263 @@
+     * @param context,  Authentication context
+     */
+    private AuthenticatorFlowStatus start(HttpServletRequest request, HttpServletResponse response,
+                                          AuthenticationContext context) throws AuthenticationFailedException {
+
+        try {
+            String userName = request.getParameter(CasqueAuthenticatorConstants.USER_NAME);
+            if (StringUtils.isEmpty(userName)) {
+                throw new CasqueException("userName is null");
+            }
+
+            context.setProperty(CasqueAuthenticatorConstants.USER_NAME, userName);
+            String tokenId = getCasqueTokenId(userName);
+            String tokenIdPlusName = tokenId + userName;
+
+            context.setProperty(CasqueAuthenticatorConstants.RADIUS_STATE, null);
+            // Initial Access Request, fixed user,  token ID + username as the password
+            // Send request as Datagram packet
+            RadiusResponse radiusResponse = Radius.sendRequest("CASQUE SNR", tokenIdPlusName, null);
+            int radiusResponseType = radiusResponse.getType();
+
+            if (radiusResponseType == RadiusResponse.ACCESS_CHALLENGE) { // Got a challenge
+                context.setProperty(CasqueAuthenticatorConstants.RADIUS_STATE, radiusResponse.getState());
+                String challenge = radiusResponse.getChallenge();
+                String contextIdentifier = context.getContextIdentifier();
+                authPages.challengePage(response, contextIdentifier, challenge);
+                return AuthenticatorFlowStatus.INCOMPLETE;
+            }
+
+            clearProperties(context);
+
+            if (radiusResponseType == RadiusResponse.ACCESS_REJECT) {
+                throw new InvalidCredentialsException(
+                        " User authentication failed due to invalid credentials ", User.getUserFromUserName(userName));
+            }
+
+            throw new InvalidCredentialsException(" User authentication failed due to " + radiusResponse.getError(),
+                    User.getUserFromUserName(userName));
+        } catch (CasqueException ce) {
+            throw new AuthenticationFailedException(ce.getMessage(), ce);
+        }
+    }
+
+    /**
+     * clear properties in case another authentication step has set some properties
+     */
+    private void clearProperties(AuthenticationContext context) {
+
+        Map<String, Object> props = new HashMap<>();
+        context.setProperties(props);
+    }
+
+    @Override
+    public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletResponse response,
+                                           AuthenticationContext context)
+            throws AuthenticationFailedException, LogoutFailedException {
+
+        if (context.isLogoutRequest()) {
+            try {
+                processLogoutResponse(request, response, context);
+            } catch (UnsupportedOperationException e) {
+                if (log.isDebugEnabled()) {
```

Comment:
here we don't need to put log.isDebugenable check

Reply:
fixed